core/src/tasks/source/destroy.ts

```diff
@@ -33,6 +33,16 @@ export class SourceDestroy extends CLSTask {
       throw error;
     }
 
+    // check if the property is directly mapped
+    try {
+      await Source.ensureDirectlyMappedPropertyNotInUse(source);
+    } catch (error) {
+      if (error.message.match(/cannot delete property/)) {
+        return; // check back later
+      }
+      throw error;
+    }
```

Comment:
We could probably add a test for this, similar to the one that checks that the source isn't being used